lib/elixir/lib/calendar/naive_datetime.ex

```diff
@@ -296,11 +360,19 @@ defmodule NaiveDateTime do
       "2000-02-29 23:00:07"
 
   """
-  @spec to_string(Calendar.naive_datetime) :: String.t
+  @spec to_string(Calendar.naive_datetime()) :: String.t()
   def to_string(naive_datetime)
 
-  def to_string(%{calendar: calendar, year: year, month: month, day: day,
-                  hour: hour, minute: minute, second: second, microsecond: microsecond}) do
+  def to_string(%{
+        calendar: calendar,
+        year: year,
+        month: month,
+        day: day,
+        hour: hour,
+        minute: minute,
+        second: second,
+        microsecond: microsecond
+      }) do
```

Comment:
Would it be better to change this to:
```elixir
def to_string(naive_datetime) do
  %{calendar: calendar,
      year: year,
      month: month,
      day: day,
      hour: hour,
      minute: minute,
      second: second,
      microsecond: microsecond} = naive_datetime

  calendar.naive_datetime_to_string(year, month, day, hour, minute, second, microsecond)
end
```

If changed and a map/struct not matching this keys is passed to the function, it would change the expected error. I felt like keeping the keys there made more sense.

comments?

Reply:
Yes please keep the keys in order to leave the semantics intact. We can always revisit later if we want to do more "refactory" changes.